File: languages/js/src/entity/identity_v2/issuer_params.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { NetworkId } from "./network";
import { Blockchain } from "./blockchain";
import { Method } from "./method";

export class IssuerParams {
    method: Method;
    blockchain: Blockchain;
    network: NetworkId;

    constructor(method: Method, blockchain: Blockchain, network: NetworkId) {
        this.method = method
        this.blockchain = blockchain
        this.network = network
    }

    public toProto(): identityEntitiesProto.IssuerParams {
        return identityEntitiesProto.IssuerParams.fromPartial({
            method: this.method,
            blockchain: this.blockchain,
            networkId: this.network
        });
    }
}

File: languages/python/bloock/entity/identity_v2/issuer_params.py
import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.method import Method
from bloock.entity.identity_v2.blockchain import Blockchain
from bloock.entity.identity_v2.network import Network


class IssuerParams:

    def __init__(self, method: Method, blockchain: Blockchain, network: Network) -> None:
        self.method = method
        self.blockchain = blockchain
        self.network = network

    def to_proto(self) -> proto.IssuerParams:
        return proto.IssuerParams(
            method=Method.to_proto(self.method),
            blockchain=Blockchain.to_proto(self.blockchain),
            network_id=Network.to_proto(self.network)
        )
